### ISSUE TITLE
Documentation request tracking

### ISSUE THREAD

**Issue by sunqm** [pyscf/pyscf#1862]
Over the past few years, we've observed that some of our documentation and examples have not received the necessary maintenance and updates. This legacy documentation has led to confusion when reading the existing code and developing new features based on existing functionalities. To help users to harness the program, more information of data structure and algorithms need to be documented in the program reference. 

This issue will collects a TODO list for updating documents and examples. Please reply to this issue for any requests in documentation updates.

* Initial guess of SCF method (#1839)
* Convention of density matrix (#1568)
* FCI wavefunction and intermediates structure (#1836, #1522)
* Custom Hamiltonian for PBC (#1819)
* Scanner for FCI solver (#1815)
* Units convention (#1814)
* CISD wavefunction (#1736, #1627)
* EOM-CC wavefunction (#1427)
* Stability analysis for GHF method (#1363)
* TDDFT results (#976)

**Comment by fishjojo** [comment]
point group symmetry conventions
different implementations of the pbc density fitting
integral / density screening strategies and cutoff parameters
many C functions in e.g., cc, ao2mo, df, ... modules (I suggest to write python wrappers for these C function calls)

**Comment by tberkel** [comment]
I agree that the User Guide will benefit from updates. Separately, it'll be
a lot of work, but someday we need to write the Developer Guide and improve
the API documentation.

**Comment by verena-neufeld** [comment]
As a partial solution, we could encourage documentation updates with related pull requests, e.g. if there is a new extension implemented or refactored, etc, the documentation could be updated, too.

**Comment by verena-neufeld** [comment]
- Listing supported operating systems. See issues #1866 , #1857 , #910 and #750 .

**Comment by sunqm** [comment]
Installation instruction for dmrgscf https://github.com/pyscf/dmrgscf/issues/7

**Comment by jeanwsr** [comment]
- [x] The [installation guide](https://pyscf.org/user/install.html#install-with-conda) should mention enabling conda-forge channel, as discussed in #2270.

**Comment by sunqm** [comment]
documents for dispersion https://github.com/pyscf/pyscf.github.io/issues/147

**Comment by jeanwsr** [comment]
Some recent new features that need documentation:
- #3049  
- #3236
- #3222, #3244 
- #3228
- #2934
- #3053, #2935
- #3420 
- #2046
- #3265
- #2480
